Replace audio backend with miniaudio (#2749)
SFML/SFML: src/SFML/Audio/CMakeLists.txt

)
source_group("codecs" FILES ${CODECS_SRC})

# Ensure certain files are compiled as Objective-C++
# See: https://miniaud.io/docs/manual/index.html#Building
if(SFML_OS_IOS)
    enable_language(OBJCXX)

>     set_source_files_properties(${SRCROOT}/AudioDevice.cpp PROPERTIES LANGUAGE OBJCXX)

Reviewer: I think I'd be okay with that since this is the kind of header-only library that requires defining its implementation in just one file.

)
source_group("codecs" FILES ${CODECS_SRC})

# Ensure certain files are compiled as Objective-C++
# See: https://miniaud.io/docs/manual/index.html#Building
if(SFML_OS_IOS)
    enable_language(OBJCXX)
    set_source_files_properties(${SRCROOT}/Miniaudio.cpp PROPERTIES LANGUAGE OBJCXX)
endif()

> # let CMake know about our additional audio libraries paths (on Android and OSX)
> if(SFML_OS_MACOSX)

Reviewer 2: ```suggestion
# let CMake know about our additional audio libraries paths (on Android and macOS)
if(SFML_OS_MACOS)
```

endif()

> # let CMake know about our additional audio libraries paths (on Android and OSX)

Reviewer 2: That part of my previous suggestion was overlooked:
```suggestion
# let CMake know about our additional audio libraries paths (on Android and macOS)
```

                 SOURCES ${SRC} ${CODECS_SRC})

> # avoids warnings in vorbisfile.h
> target_compile_definitions(sfml-audio PRIVATE OV_EXCLUDE_STATIC_CALLBACKS FLAC__NO_DLL MA_NO_DECODING MA_NO_ENCODING)

Reviewer 2: ```suggestion
# avoids warnings in vorbisfile.h
target_compile_definitions(sfml-audio PRIVATE OV_EXCLUDE_STATIC_CALLBACKS FLAC__NO_DLL)
# disables miniaudio features we do not use
target_compile_definitions(sfml-audio PRIVATE MA_NO_DECODING MA_NO_ENCODING)
```
